qubesadmin/tools/qvm_template.py

```diff
@@ -539,7 +539,7 @@ def qrexec_repoquery(
         for line in stderr.rstrip().split('\n'):
             print(f"[Qrexec] {line}", file=sys.stderr)
         raise ConnectionError("qrexec call 'qubes.TemplateSearch' failed.")
-    name_re = re.compile(r'\A[A-Za-z0-9._+][A-Za-z0-9._+-]*\Z')
+    name_re = re.compile(r'\A[A-Za-z0-9._+][A-Za-z0-9. _+-]*\Z')
```

Comment:
I haven't checked other places yet, but space is not a valid character in a template name. Is it leading/trailing space? If so, maybe `strip()` call is needed somewhere?

Reply:
> I haven't checked other places yet, but space is not a valid character in a template name. Is it leading/trailing space? If so, maybe `strip()` call is needed somewhere?

It is neither leading or trailing space. The unittest has the actual output data from DNF5. It was:
`Qubes Templates repository`

Reply:
Ok, so this is repository name, not a template name. The `qubes.TemplateSearch ` should have repository id here, not its name (I see variable is named `reponame`, but that's not accurate). So, it either needs an option for dnf to print repo id, or a mapping to translate repo name to repo id. This change belongs to the core-agent-linux PR.

Reply:
> Ok, so this is repository name, not a template name. The `qubes.TemplateSearch ` should have repository id here, not its name (I see variable is named `reponame`, but that's not accurate). So, it either needs an option for dnf to print repo id, or a mapping to translate repo name to repo id. This change belongs to the core-agent-linux PR.

Ok. I will remove space from the Regex. I am not certain if an option could be found to behave the same data in both DNF4 & DNF5. The [current existing key](https://github.com/QubesOS/qubes-core-agent-linux/blob/main/qubes-rpc/qvm-template-repo-query#L52C76-L52C88) in `qvm-template-repo-query` is `%{reponame}` which behaves different in DNF4 and DNF5.

Reply:
Maybe there is `%{repoid}` or similar that would work on DNF5? (or ideally also with DNF4?) it can use different options for DNF4 and DNF5

Reply:
> Maybe there is `%{repoid}` or similar that would work on DNF5? (or ideally also with DNF4?) it can use different options for DNF4 and DNF5

Just checked it. `%{repoid}` is valid for both DNF4 and DNF5. Fix for `core-agent-linux` here:

https://github.com/alimirjamali/qubes-core-agent-linux/pull/new/DNF5-repoquery-fix

And this PR is also updated